cranelift/codegen/src/isa/x64/lower.rs

```diff
@@ -1732,6 +1748,148 @@ fn lower_insn_to_regs<C: LowerCtx<I = Inst>>(
             });
         }
 
+        Opcode::AtomicRmw => {
+            // This is a simple, general-case atomic update, based on a loop involving
+            // `cmpxchg`.  Note that we could do much better than this in the case where the old
+            // value at the location (that is to say, the SSA `Value` computed by this CLIF
+            // insn) is not required.  In that case, we could instead implement this using a
+            // single `lock`-prefixed x64 read-modify-write instruction.  Also, even in the case
+            // where the old value is required, for the `add` and `sub` cases, we can use the
+            // single instruction `lock xadd`.  However, those improvements have been left for
+            // another day.
```

Comment:
Can you open an issue for the improvements, please, and refer to it with a `TODO` comment mentioning the issue number?